packages/paste-core/components/disclosure/src/DisclosureHeading.tsx

```diff
@@ -1,77 +1,114 @@
+    if (isDisabled) shouldIconMove = isExpanded ? true : false;
+
+    let variantStyles: BoxStyleProps = {};
     if (disclosureVariant === 'contained') {
-      bottomBorderRadius = 'borderRadius10';
-    }
-    if (disclosureVariant === 'contained' && props['aria-expanded']) {
-      bottomBorderRadius = 'borderRadius0';
+      variantStyles = isHovered ? containedHoverStyles : containedStyles;
+      if (isExpanded) {
+        variantStyles = {
+          ...variantStyles,
+          borderBottomLeftRadius: 'borderRadius0',
+          borderBottomRightRadius: 'borderRadius0',
+          borderBottom: 'none',
+        };
+      }
     }
 
     return (
       <Heading as={renderAs} marginBottom="space0" variant={variant} ref={ref}>
         <Box
           {...safelySpreadBoxProps(props)}
           as="div"
-          backgroundColor={props['aria-expanded'] ? 'colorBackground' : 'colorBackgroundBody'}
-          borderRadius={disclosureVariant === 'contained' ? 'borderRadius10' : 'borderRadius20'}
-          borderBottomLeftRadius={bottomBorderRadius}
-          borderBottomRightRadius={bottomBorderRadius}
+          backgroundColor={isExpanded ? 'colorBackgroundWeak' : 'colorBackgroundBody'}
+          borderRadius="borderRadius20"
           cursor="pointer"
           display="flex"
           element={element}
           outline="none"
           padding="space30"
           position="relative"
-          ref={hoverRef}
+          ref={buttonRef}
           role="button"
           zIndex="zIndex10"
-          transition="background-color 100ms ease-out"
+          transition="background-color, border-color, border-radius 100ms ease"
           _hover={{
-            backgroundColor: 'colorBackgroundStrong',
+            backgroundColor: 'colorBackground',
           }}
           _focus={{
             boxShadow: 'shadowFocus',
           }}
           _disabled={{
-            backgroundColor: props['aria-expanded'] ? 'colorBackground' : 'colorBackgroundBody',
-            color: 'colorTextWeak',
+            backgroundColor: 'colorBackgroundStrong',
+            color: 'colorTextWeaker',
             cursor: 'not-allowed',
           }}
+          {...variantStyles}
         >
           <Box
             as="span"
             element={`${element}_ICON`}
             display="flex"
-            height={IconSizeFromHeading[variant] || 'sizeIcon20'}
-            width={IconSizeFromHeading[variant] || 'sizeIcon20'}
-            {...getIconHoverStyles(isHovered, variant, props[`aria-expanded`], props[`aria-disabled`], theme.space)}
+            color={isDisabled ? 'colorTextWeaker' : 'colorTextIcon'}
+            transition="transform 200ms ease-out"
+            height={iconSize}
+            width={iconSize}
+            transform={`
+            ${shouldIconMove ? `translateX(${theme.space.space10})` : `translateX(0)`}
+            ${isExpanded ? 'rotate(90deg)' : 'rotate(0deg)'}`}
```

Comment:
issue: When the Disclosure is expanded, the icon should move down 2px. Basically, same idea as when it's collapsed, but along Y instead of X.

Reply:
I noticed this as well

Reply:
@shleewhite I'm still not seeing the chevron moving down 2px on hover when Disclosure is expanded. 
I do see that the icon is overlapping with the heading a bit (see screenshot). I'm not exactly sure how to fix it, but I removed the translate values and it seemed to help with the overlapping issue, but not with the hover state animation. 

<img width="1200" alt="Screenshot 2023-01-24 at 8 56 55 AM" src="https://user-images.githubusercontent.com/1484014/214337893-25f2277b-5bff-4f34-a87a-6f2b36a912d5.png">